### PR TITLE
Enable dependency isolation by default for python 3.10

### PR DESCRIPTION
**Pull request by gavin-aguiar**
## Description
Enabling capability to separate customer's site-packages from worker's dependencies for python 3.10

Fixes #

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/unittests/test_utilities_dependency.py [hunk 1]
```diff
@@ -558,11 +555,11 @@ def test_use_worker_dependencies_disable(self):
...
 
     @unittest.skipUnless(
-        sys.version_info.major == 3 and sys.version_info.minor in (6, 7, 8),
-        'Test only available for Python 3.6, 3.7, or 3.8'
+        sys.version_info.major == 3 and sys.version_info.minor in (6, 7, 8, 9),
+        'Test only available for Python 3.6, 3.7, 3.8 or 3.9'
     )
-    def test_use_worker_dependencies_default_python_36_37_38(self):
-        # Feature should be disabled in Python 3.6, 3.7, and 3.8
+    def test_use_worker_dependencies_default_python_36_37_38_39(self):
```
Comment (vrdmr) [inline tests/unittests/test_utilities_dependency.py:561]: Why do we need to explicitly add this check? Why not do a simpler check

`sys.version_info.major == 3 and sys.version_info.minor != 10`

Comment (vrdmr) [reply tests/unittests/test_utilities_dependency.py:561]: Or better, use the below option of skipIf

```python
@unittest.skipIf(sys.version_info.major == 3 and sys.version_info.minor == 10,
                 'Test not available for python 3.10')
```

**Review by vrdmr — APPROVED** [review]
- nit changes.
LGTM

File: tests/unittests/test_utilities_dependency.py (hunk 1 above)
Comment (gavin-aguiar) [reply tests/unittests/test_utilities_dependency.py:561]: This check was to make sure the dependency isolation is not enabled for other versions. Thats why I didnt use a skipif. Changed the condition to !=10.

**Pull request merged**